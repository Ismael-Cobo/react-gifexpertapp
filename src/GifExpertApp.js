import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'


export const GifExpertApp = () => {

    //const categories = ['One punch', 'Samurai X', 'Dragon ball']
    const [categories, setcategories] = useState([''])

    /*const handleAdd = () => {
        setcategories([...categories, 'hunter x hunter'])
        //setcategories(category => [category, 'HunterXHunter'])
    }*/

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setcategories = {setcategories} />
            
            <hr />
            
            <ol>
                {
                    categories.map(category => (
                        <GifGrid 
                            key={category} 
                            category={category} 
                        />
                    ))
                }
            </ol>
        </>
    )
}
