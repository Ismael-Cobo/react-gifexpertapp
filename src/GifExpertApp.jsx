import { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

import './styles/styles.css'


export const GifExpertApp = ({ defaultValue = [] }) => {

    // const [categories, setCategories] = useState(['one punch man'])
    const [categories, setCategories] = useState(defaultValue)


    return (
        <div className='gif__contenedor'>
            <h3 className='gif__titulo'>GifExpertApp</h3>
            <AddCategory setCategories={setCategories} />
            {/* Aqui mapeo el conjunto de items con el grid */}
            <ol className='gif__ol'>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </div>
    )
}
