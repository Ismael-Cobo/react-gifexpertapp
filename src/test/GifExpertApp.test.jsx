import { shallow } from "enzyme"
import { GifExpertApp } from "../GifExpertApp"

describe('pruebas en <GifExpertApp />', () => {

    test('debe mostrar el componente correctamente', () => {

        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot()
    })

    test('debe de mostrar una lista del componente <GifGrid />', () => {

        const values = [
            'one punch',
            'dragon-ball'
        ]
        const wrapper = shallow(<GifExpertApp defaultValue={values} />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(values.length)
    })
})