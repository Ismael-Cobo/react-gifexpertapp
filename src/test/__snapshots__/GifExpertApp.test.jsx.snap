// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp /> debe de mostrar una lista del componente <GifGrid /> 1`] = `
<div
  className="gif__contenedor"
>
  <h3
    className="gif__titulo"
  >
    GifExpertApp
  </h3>
  <AddCategory
    setCategories={[Function]}
  />
  <ol
    className="gif__ol"
  >
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="dragon-ball"
      key="dragon-ball"
    />
  </ol>
</div>
`;

exports[`pruebas en <GifExpertApp /> debe mostrar el componente correctamente 1`] = `
<div
  className="gif__contenedor"
>
  <h3
    className="gif__titulo"
  >
    GifExpertApp
  </h3>
  <AddCategory
    setCategories={[Function]}
  />
  <ol
    className="gif__ol"
  />
</div>
`;
